workflow/scripts/osemosys_global/transmission/main.py

```diff
@@ -0,0 +1,247 @@
+    activity_transmission_limit,
+    capact_transmission
+    )
+
+from costs import get_transmission_costs
+
+from operational_life import set_op_life_transmission
+
+from investment_constraints import cap_investment_constraints_trn
+
+from user_defined_capacity import set_user_defined_capacity_trn
+
+from sets import create_tech_set_trn
+
+def main(
+    plexos_prop: pd.DataFrame,
+    default_op_life: pd.DataFrame,
+    line_data: pd.DataFrame,
+    interface_data: pd.DataFrame,
+    iar_base: pd.DataFrame,
+    oar_base: pd.DataFrame,
+    capact_base: pd.DataFrame,
+    cap_cost_base: pd.DataFrame,
+    fix_cost_base: pd.DataFrame,
+    op_life_base: pd.DataFrame,
+    max_cap_invest_base: pd.DataFrame,
+    min_cap_invest_base: pd.DataFrame,
+    res_cap_base: pd.DataFrame,
+    tech_set_base: pd.DataFrame,
+):
+    
+    # CALL FUNCTIONS
+    
+    # Set activity ratios for transmission.
+    df_iar_trn_final, df_oar_trn_final = activity_transmission(iar_base, oar_base, 
+                                                               plexos_prop, interface_data,
+                                                               start_year, end_year,
+                                                               region_name)
+
+    # Adjust activity limits if cross border trade is not allowed following user config.
+    df_crossborder_final = activity_transmission_limit(cross_border_trade, df_oar_trn_final)
+    
+    # Set CapacityToActivityUnit.
+    df_capact_trn_final = capact_transmission(capact_base, df_oar_trn_final)
+    
+    # Set capital and fixed transmission costs.
+    df_cap_cost_trn_final, df_fix_cost_trn_final = get_transmission_costs(line_data, df_oar_trn_final,
+                                                                    cap_cost_base, fix_cost_base,
+                                                                    start_year, end_year,
+                                                                    region_name)
+
+    # Set operational life for transmission.
+    df_op_life_trn_final = set_op_life_transmission(df_iar_trn_final, df_oar_trn_final, 
+                                                    op_life_dict, op_life_base, region_name)
+    
+    # Set annual capacity investment constraints.
+    df_max_cap_invest_trn_final = cap_investment_constraints_trn(df_iar_trn_final, 
+                                                                 max_cap_invest_base, 
+                                                                 no_investment_techs, 
+                                                                 start_year, 
+                                                                 end_year, 
+                                                                 region_name)
+    
+    tech_set_trn_final = pd.concat([tech_set, create_tech_set_trn(interface_data)])
+
+    # Alter output csv's based on user defined capacities following user config.
+    if not tech_capacity_trn is None:
```

Comment:
Kinda pedantic of me here, but PEP 8 recommends not using this syntax (see third bullet point [here](https://peps.python.org/pep-0008/#programming-recommendations)). Rather, PEP8 suggests
```python 
if tech_capacity_trn is not None:
  ...
```

Not requiring this change, just a FYI :)